metadata-ingestion/src/datahub/ingestion/source/sql/mssql/source.py

```diff
@@ -237,6 +270,343 @@ def create(cls, config_dict: Dict, ctx: PipelineContext) -> "SQLServerSource":
+
+                IF @referenced_object_type = 'SN'
+                  BEGIN
+                      BEGIN TRY
+                          SET @SQL  = 'SELECT
+                                      ' + CAST(@id as NVARCHAR(10)) + '
+                                        , syn_obj.name
+                                        , syn_obj.type
+                                        , syn_obj.object_id
+                                      FROM ' + COALESCE(QUOTENAME(@referenced_database_name), '') + '.sys.synonyms AS synon
+                                      INNER JOIN ' + COALESCE(QUOTENAME(@referenced_database_name), '') + '.sys.objects  AS syn_obj
+                                            ON OBJECT_ID(synon.base_object_name) = syn_obj.object_id
+                                      WHERE synon.object_id = ' + CAST(@referenced_id AS NVARCHAR(10)) + ';';
+                            PRINT @SQL;
+                          INSERT INTO @synonym_info(
+                              id
+                            , referenced_name
+                            , referenced_type
+                            , referenced_object_id
+                              )
+                          EXEC sp_executesql @SQL;
+
+                          UPDATE #ProceduresDependencies
+                          SET
+                              referenced_object_type = si.referenced_type,
+                              referenced_entity_name = si.referenced_name,
+                              referenced_id = si.referenced_object_id
+                          FROM #ProceduresDependencies pd
+                          INNER JOIN @synonym_info si
+                              ON si.id = pd.id
+                          WHERE CURRENT OF proc_depend_cursor;
+                      END TRY
+                      BEGIN CATCH
+                          PRINT ERROR_MESSAGE()
+                      END CATCH;
+                  END;
+
+                FETCH NEXT FROM proc_depend_cursor INTO
+                    @id
+                  , @referenced_database_name
+                  , @referenced_id
+                  , @referenced_object_type;
+            END;
+            CLOSE proc_depend_cursor;
+            DEALLOCATE proc_depend_cursor;
+            END;
+            """
+        )
+
+        _dependencies = conn.execute(
+            """
+            SELECT DISTINCT
+                *
+            FROM #ProceduresDependencies
+            WHERE referenced_object_type IN ('U ', 'V ', 'P ');
+            """
+        )
+
+        # PAY ATTENTION:
+        # at 19.07.2024 self.get_db_name method converts db name to lowercase forcely:
+        # return str(engine.url.database).strip('"').lower()
+        # to avoid inconsistency, referenced_database_name in this module repeats the same behavior
+        # remove after fixing
+        db_force_converting = (
+            True if "lower()" in inspect_objects.getsource(self.get_db_name) else False
```

Comment:
this seems extremely hacky, and it's not clear to me why this is necessary?

Reply:
MSSQL [db_name()](https://learn.microsoft.com/en-us/sql/t-sql/functions/db-name-transact-sql?view=sql-server-ver16) function, that is used in qury of extraction dependencies, returns name of DB as it was declared. If name of DB was constructed in CamelCase or any uppercase symbols exist, function returns value as is:
![изображение](https://github.com/user-attachments/assets/5e082e8a-641b-455c-b8c8-620c0bb5a093)


But [get_db_name](https://github.com/datahub-project/datahub/blob/66ecfae5e48b27c8477ccec94b82db8a51322811/metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py#L411)  contains logic of transforming name to lowercase style. To avoid this inconsistency, this small check is implemented.

Reply:
now that https://github.com/datahub-project/datahub/pull/10773 has been merged, can we remove this?